### PR TITLE
Support FreeBSD

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by mmoll** [comment]
also removes leftovers of OSX support and rearranges params.pp

File: manifests/init.pp [hunk 1]
```diff
@@ -69,14 +70,22 @@
         content => template('dhcp/redhat/sysconfig-dhcpd'),
       }
     }
-    default: {
-      fail("Unsupported OS family ${::osfamily}")
```
Comment (ekohl) [inline manifests/init.pp]: Why remote this line?

File: manifests/init.pp [hunk 2]
```diff
@@ -69,14 +70,22 @@
...
+    /^(FreeBSD|DragonFly)$/: {
+      $interfaces_line = inline_template ("<%= @dhcp_interfaces.join(' ') -%>")
+      augeas { 'set listen interfaces':
+        context => '/files/etc/rc.conf',
+        changes => "set dhcpd_ifaces '\"${interfaces_line}\"'",
+        before  => Package[$packagename],
+        notify  => Service[$servicename],
+      }
+    }
+  default: {
```
Comment (ekohl) [inline manifests/init.pp]: Odd indenting?

File: manifests/init.pp (hunk 1 above)
Comment (mmoll) [inline manifests/init.pp]: because it's already failing in params.pp

File: manifests/init.pp (hunk 2 above)
Comment (mmoll) [inline manifests/init.pp]: fixed :)

**Comment by domcleal** [comment]
:+1:

**Comment by ekohl** [comment]
If you're removing darwin from the list, then you can also remove ```$package_provider```.

**Comment by mmoll** [comment]
indeed, done

File: manifests/init.pp [hunk 3]
```diff
@@ -69,14 +64,22 @@
         content => template('dhcp/redhat/sysconfig-dhcpd'),
       }
     }
+    /^(FreeBSD|DragonFly)$/: {
+      $interfaces_line = inline_template ("<%= @dhcp_interfaces.join(' ') -%>")
```
Comment (ekohl) [inline manifests/init.pp]: I'd prefer pure puppet (```$interfaces_line = join($dhcp_interfaces, ' ')```) or move the template to ```changes =>``` below

Comment (mmoll) [inline manifests/init.pp]: fixed, I just blindly copied from the template...

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**